config/config.go

```diff
@@ -127,6 +129,17 @@ func WithOpenTelemetryConfiguration(cfg OpenTelemetryConfiguration) Configuratio
 	})
 }
 
+// WithPrometheusRegisterer sets a Prometheus registerer to be used by
+// any Prometheus exporters configured in this SDK. Note: if this option
+// is set, the caller is expected to initialize their own handler to
```

Comment:
Can you help explain this a bit more to me?  Doesn't this mean most of the prometheus configuration provided via config (e.g. port, path) is ignored?  Why does the collector need this behavior?

Reply:
So the bulk of the prometheus handler code in the collector exists here: https://github.com/open-telemetry/opentelemetry-collector/blob/e99074da2f6525cd177f26b46b8d03fef2fe4f35/service/internal/proctelemetry/config.go#L106-L116

I'm totally open to suggestions as to a better way to implement this, the problem I was trying to solve was handling errors without passing a channel around.

Reply:
So the error channel used terminates the collector on error:

```
    // AsyncErrorChannel is the channel that is used to report fatal errors.
    AsyncErrorChannel chan error
```

I suppose we could register an error handler and set it to return the error to the channel instead?

Reply:
I'll try to implement that and see how far i can get. will put this in draft until I get that working